vscode/src/workspace.ts

```diff
@@ -373,7 +373,7 @@ export class Workspace implements WorkspaceInterface {
 
     // If a configuration that affects the Ruby LSP has changed, update the client options using the latest
     // configuration and restart the server
-    this.context.subscriptions.push(
+    this.subscriptions.push(
```

Comment:
Noob question: why don't we push these to the client too and then let its `dispose` handle them?

Reply:
Because you can restart the LSP client as many times as you like in the same workspace, but we never re-create the file watchers (disposables). We can only dispose of them when the workspace is being removed/deactivated.

Reply:
Yes. Added some comments, let me know if you think it's clear.